### PR TITLE
[Demo] Add Example DAC GitHub Action CI Workflows

### PR DESCRIPTION
**Pull request by Mikaayenson**
## Summary

Adds three different example DAC GitHub Action CI Workflows:

1. On merge to DAC-feature (this is production branch in this repo)
 - Push to `prod` space
2. On PR open/commit (any)
 - Push to `dev` space
3. Manual Dispatch Job (manual Sync)
 - Pull rules from Kibana compare versions to Repo

### CODE REVIEW

File: .github/workflows/pull_latest_from_elastic_security.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      run: |
+        python -m pip install --upgrade pip
+        pip cache purge
+        pip install .[dev]
+
+    - name: Export and Import Rules if flag is true
+      if: ${{ github.event.inputs.sync_rules_from_elastic == 'true' }}
+      run: |
+        # TODO: default to CUSTOM_RULES_DIR and remove --directory flags
+        python -m detection_rules kibana export-rules  --directory ${{ env.CUSTOM_RULES_DIR }}
```
Comment (Mikaayenson) [inline .github/workflows/pull_latest_from_elastic_security.yml]: specify space

File: .github/workflows/pull_latest_from_elastic_security.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        body: "This PR syncs rules from Elastic Security to the repository for review."
+        branch: "sync-rules-${{ github.run_id }}"
+        delete-branch: true
+        labels: "auto-update"
+
+    - name: Commit Directly to Main
+      if: ${{ github.event.inputs.sync_rules_from_elastic == 'false' }}
+      run: |
+        git config --global user.name 'GitHub Action'
+        git config --global user.email 'action@github.com'
```
Comment (Mikaayenson) [inline .github/workflows/pull_latest_from_elastic_security.yml]: Update

File: .github/workflows/sync_rule_tuning_to_elasic_security_dev.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip cache purge
+        pip install .[dev]
+
+    - name: Import Specific Rules Modified in PR
+      run: |
+        find "${{ env.CUSTOM_RULES_DIR }}" -name '*.toml' | while read rule_file; do
```
Comment (Mikaayenson) [inline .github/workflows/sync_rule_tuning_to_elasic_security_dev.yml]: Should only be changed rules

File: .github/workflows/pull_latest_from_elastic_security.yml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    - name: Update Version Lock
+      run: |
+        python -m detection_rules dev build-release --update-version-lock
+
+    - name: Create Pull Request
+      if: ${{ github.event.inputs.sync_rules_from_elastic == 'true' }}
+      uses: peter-evans/create-pull-request@v6
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (Mikaayenson) [inline .github/workflows/pull_latest_from_elastic_security.yml]:
```suggestion
        token: ${{ secrets.WRITE_GITHUB_TOKEN }}
```

File: .github/workflows/sync_rule_tuning_to_elasic_security_dev.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        python -m pip install --upgrade pip
+        pip cache purge
+        pip install .[dev]
+
+    - name: Import Specific Rules Modified in PR
+      run: |
+        git fetch origin ${{ github.base_ref }}
+        git diff --name-only ${{ github.event.before }} ${{ github.event.after }} '${{ env.CUSTOM_RULES_DIR }}/**/*.toml' | while read rule_file; do
+          echo "Importing rule from file $rule_file"
+          python -m detection_rules kibana import-rules --rule-file "$rule_file" --space "dev"
```
Comment (eric-forte-elastic) [inline .github/workflows/sync_rule_tuning_to_elasic_security_dev.yml]:
```suggestion
          python -m detection_rules kibana import-rules --rule-file "$rule_file" --space "dev" --overwrite
```

Comment (eric-forte-elastic) [reply .github/workflows/sync_rule_tuning_to_elasic_security_dev.yml]: I think?

**Review by eric-forte-elastic — APPROVED** [review]
:green_circle: Peer review, looks good to me! :+1:

**Pull request merged**